src/handler/patch.ts

```diff
@@ -0,0 +1,64 @@
+import 'dotenv/config';
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import logger from '../util/logger';
+import { addHttpHeaders } from '../util/httpHeaders';
+import {
+  archiveOldCreateCurrentRecord,
+  getBySystemNumberAndCreatedTimestamp,
+} from '../services/database';
+import { validateUpdateVinRequest, validateVins } from '../validators/patch';
+import { PatchRequestRecords } from '../processors/processPatchVinRequest';
+import { TechrecordGet } from '../models/post';
+
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+  logger.info('Patch Technical Record Called');
+
+  const isRequestInvalid = validateUpdateVinRequest(event);
+
+  if (isRequestInvalid) {
+    return isRequestInvalid;
+  }
+
+  logger.info('Request is Valid');
+
+  const systemNumber: string = decodeURIComponent(
```

Comment:
is there a need to decode these?

Reply:
This was behaviour from the Get endpoint. My gut feeling is no, because I don't think ISO strings should contain unsafe characters and system numbers are I think alpha numeric.